tests/rptest/tests/flink_basic_test.py

```diff
@@ -231,59 +229,57 @@ def test_transaction_workload(self):
+        def list_files(path):
+            files = []
             try:
+                # Note that there is no need to check file name as tmp file will
+                # have filename started with '.part' and it will not be listed by
+                # bare 'ls -1' command
                 files = self.flink.node.account.ssh_output(
                     f"ls -1 {_data_path}")
                 files = files.decode().splitlines()
             except RemoteCommandError:
                 # Ignore ssh_output command errors
                 # as folder will not be existent just yet
-                continue
-            # exit if file is present
-            # Not that there is no need to check file name as tmp file will
-            # have filename started with '.part' and it will not be listed by
-            # bare 'ls -1' command
-            if len(files) > 0:
-                break
-            self.logger.info("No data file present. "
-                             f"Sleeping for {backoff_sec} sec")
-            time.sleep(backoff_sec)
-            iterations -= 1
-
-        # There should be at least one file
-        assert len(
-            files
-        ) > 0, f"Flink transaction workload produced no data files after 5 min"
+                pass
+            return files
+
+        # Wait for file, timeout 5 min (10 sec * 30 iterations)
+        self.logger.info("Waiting for consume workload data file")
+        wait_until(lambda: len(list_files(_data_path)) > 0,
+                   timeout_sec=300,
+                   backoff_sec=10,
+                   err_msg="Flink transaction workload produced"
+                   "no data files after 5 min")
 
         # Load data file and parse it
-        data = {}
-        for f in files:
-            # Create record
-            if f not in data:
-                data[f] = []
-            # This is basic test, no big data expected
-            # So, no copying file locally.
-            csv_data = self.flink.node.account.ssh_output(
-                f"cat {_data_path}/{f}").decode()
-            # Parse the csv
-            # Second parameter is the quick and dirty value type map
-            data[f] = self._parse_csv(csv_data, [int, int, int, str, int, str])
-
-        # Find max offset
-        offset_column = 1
+        # For most efficient way of processing data it would be more
+        # convenient to copy script to target node and run it there
+        # But this is basic test, and copying file locally is a more
+        # simple approach in this case.
+
+        # Max index and target column
+        index_column = 1
         max_index = 0
-        for ff, part_data in data.items():
-            for row in part_data:
-                if max_index < row[offset_column]:
-                    max_index = row[offset_column]
+        for f in list_files(_data_path):
+            # Copy file locally
+            tmpfile = os.path.join("/tmp", "tmpcsvdata")
```

Comment:
nit: use tempfile module

with tempfile.NamedTemporaryFile(suffix=".csv") as tmpfile

cleans up in all cases and removes hard coded dependency on tmp

Reply:
Got it. Will update with next PR.